pkg/build/api/types.go

```diff
@@ -707,6 +771,28 @@ type BuildRequest struct {
 
 	// Env contains additional environment variables you want to pass into a builder container
 	Env []kapi.EnvVar
+
+	// BuildCauses can pass information from the controllers to the buildconfig
+	// generator about who/what has triggered a build through the build request.
+	BuildCauses []*BuildTriggerCauses
```

Comment:
Let's assume a single Cause for now.  If we ever end up with level driven builds and we decide that means we want to list multiple causes ("this build was triggered because image foo changed and also commit X was pushed"), we'll iterate on the api at that point.  I don't want to sign up for that complexity at the moment.